pool/context_pool_test.go

```diff
@@ -113,6 +113,22 @@ func TestContextPool(t *testing.T) {
 			})
 			require.ErrorIs(t, p.Wait(), context.DeadlineExceeded)
 		})
+
+		t.Run("return before timed out", func(t *testing.T) {
+			t.Parallel()
+			ctx, cancel := context.WithTimeout(bgctx, 2*time.Millisecond)
+			defer cancel()
+			p := New().WithContext(ctx)
+			p.Go(func(ctx context.Context) error {
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				case <-time.After(1 * time.Millisecond):
+					return nil
+				}
+			})
+			require.NoError(t, p.Wait())
+		})
```

Comment:
Hey! I was just going to merge this (sorry for the delay), but noticed that this test case is inherently racy. Basically, we can't know for sure that the goroutine will be scheduled before the context times out, so `time.After()` may trigger _after_ `ctx.Done()`. 

I've pushed a commit that uses a context that does not time out instead.